components/provider_ppucd/src/structs.rs

```diff
@@ -0,0 +1,156 @@
+    pub const NFDINERT_V1: DataKey = data_key!(uniset, "nfdinert", 1);
+    pub const NFKCINERT_V1: DataKey = data_key!(uniset, "nfkcinert", 1);
+    pub const NFKDINERT_V1: DataKey = data_key!(uniset, "nfkdinert", 1);
+    pub const PAT_SYN_V1: DataKey = data_key!(uniset, "Pat_Syn", 1);
+    pub const PAT_WS_V1: DataKey = data_key!(uniset, "Pat_WS", 1);
+    pub const PCM_V1: DataKey = data_key!(uniset, "PCM", 1);
+    pub const PRINT_V1: DataKey = data_key!(uniset, "print", 1);
+    pub const QMARK_V1: DataKey = data_key!(uniset, "QMark", 1);
+    pub const RADICAL_V1: DataKey = data_key!(uniset, "Radical", 1);
+    pub const RI_V1: DataKey = data_key!(uniset, "RI", 1);
+    pub const SD_V1: DataKey = data_key!(uniset, "SD", 1);
+    pub const SEGSTART_V1: DataKey = data_key!(uniset, "segstart", 1);
+    pub const SENSITIVE_V1: DataKey = data_key!(uniset, "Sensitive", 1);
+    pub const STERM_V1: DataKey = data_key!(uniset, "STerm", 1);
+    pub const TERM_V1: DataKey = data_key!(uniset, "Term", 1);
+    pub const UIDEO_V1: DataKey = data_key!(uniset, "UIdeo", 1);
+    pub const UPPER_V1: DataKey = data_key!(uniset, "Upper", 1);
+    pub const VS_V1: DataKey = data_key!(uniset, "VS", 1);
+    pub const WSPACE_V1: DataKey = data_key!(uniset, "WSpace", 1);
+    pub const XDIGIT_V1: DataKey = data_key!(uniset, "xdigit", 1);
+    pub const XIDC_V1: DataKey = data_key!(uniset, "XIDC", 1);
+    pub const XIDS_V1: DataKey = data_key!(uniset, "XIDS", 1);
+}
+
+/// Gets a locale-invariant default struct given a data key in this module's category.
+#[cfg(feature = "invariant")]
+pub(crate) fn get_invariant(data_key: &DataKey) -> Option<DataResponse<'static>> {
+    use crate::invariant::make_inv_response;
+    match *data_key {
+        key::WSPACE_V1 => make_inv_response::<UnicodeProperty>(),
+        _ => None,
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
+pub struct UnicodeProperties {
+    pub props: Vec<UnicodeProperty>,
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
+pub struct UnicodeProperty {
+    pub name: String,
+    pub inv_list: Vec<u32>,
+}
+
+impl UnicodeProperty {
+    /// Default empty nameless property
+    pub fn default() -> UnicodeProperty {
+        UnicodeProperty {
+            name: String::new(),
+            inv_list: vec![],
+        }
+    }
+
+    /// Converts a UnicodeSet into a corresponding UnicodeProperty struct
+    /// for serde JSON de-/serialization.
+    pub fn from_uniset(s: &UnicodeSet, name: &str) -> UnicodeProperty {
+        let inv_list = s.get_inversion_list();
+        UnicodeProperty {
+            name: String::from(name),
+            inv_list,
+        }
+    }
+}
+
+impl TryInto<UnicodeSet> for UnicodeProperty {
+    type Error = crate::error::Error;
+    fn try_into(self) -> Result<UnicodeSet, Self::Error> {
+        UnicodeSet::from_inversion_list(self.inv_list)
+            .map_err(crate::error::Error::UnisetConversion)
```

Comment:
Just putting the import at the top with `use` might be preferable incase there is future need